tests/e2e/testdata/scenarios/direct/llm_immutable_field/labels_value_type/_object01.yaml

```diff
@@ -40,18 +40,9 @@ spec:
 status:
   conditions:
   - lastTransitionTime: "1970-01-01T00:00:00Z"
-    message: |-
-      Update call failed: error updating: logMetric update failed: googleapi: Error 400: Cannot change value type of label: testkey
-      Details:
-      [
-        {
-          "@type": "type.googleapis.com/google.rpc.DebugInfo",
-          "detail": "[ORIGINAL ERROR] generic::invalid_argument: Cannot change value type of label: testkey [google.rpc.error_details_ext] { message: \"Cannot change value type of label: testkey\" }"
-        }
-      ]
-      , badRequest
-    reason: UpdateFailed
-    status: "False"
+    message: The resource is up to date
+    reason: UpToDate
+    status: "True"
```

Comment:
This change is unexpected... we would expect an error message rather than "UpToDate". Have you rebased with #1813?

Reply:
Yes it is rebased and good catch! I had a similar moment of reflection here.

The caveat is that the test was captured against `real` AND as a DCL controller whereas this is being captured against `mock` and as a direct controller.

So this is where we are with the `direct` controller against `mock` ... that means we need to improve the mock! You can also see that the `export` is changed but that shouldn't happen for the `label`. 

If it's ok with you, could I improve the mock as a follow on ?